data/helpers.d/postgresql

```diff
@@ -187,7 +187,12 @@ ynh_psql_database_exists() {
     # Manage arguments with getopts
     ynh_handle_getopts_args "$@"
 
-    if ! sudo --login --user=postgres PGUSER="postgres" PGPASSWORD="$(cat $PSQL_ROOT_PWD_FILE)" psql -tAc "SELECT datname FROM pg_database WHERE datname='$database';" | grep --quiet "$database"
+    # if psql is not there, we cannot check the db
+    # though it could exists.
+    if ! command -v psql
+    then
+      ynh_die --message="PostgreSQL is not installed, impossible to check for db existence"
```

Comment:
Uuuuh I'm wondering if this helper ynh_psql_database_exists is called directly or indirectly during remove script, that could exit the remove script prematurely..?

Reply:
yes you're right ("If it can happen..."). OMG I don't know if there is a way out of this then.

Reply:
I'm guessing you had a real-life context that led you to implement this, soooo uuuh what was it ? :P 

Naively just adding a warning could be good enough ?

Reply:
Well it's exactly what I wrote actually : I've decided to put all the checks (does the user/home dir/db already exists?) at the beginning of my script. Then of course it failed because I didn't have psql yet. I just thought that it should log correctly in this case, because I felt like it took me too much time to check what was happening.

But you're right : I should keep the old behaviour but just add a warning. I'm not yet sure how bash behaves in this case though... (what does it do? Returns 0? Returns nothing?)

Reply:
Hmmmm logging a warning is just a message and is idependant of what the function returns. If you want it to return 1, then just write `return 1`

Reply:
In addition `ynh_psql_database_exists` seems to only be use in `ynh_psql_remove_db` that is only used in the removal script... My opinion is: no need to check if postgresl is installed or not

Reply:
> In addition `ynh_psql_database_exists` seems to only be use in `ynh_psql_remove_db` that is only used in the removal script... My opinion is: no need to check if postgresl is installed or not

If I read correctly the code of `ynh_psql_create_db`, I think we should alway call this function before attempting to create a db, only for the fact that `ynh_psql_create_db` modifies owner and rights quite violently. I actually do this check in `diaspora_ynh`. (actually maybe another fix should be to make ynh_psql_create_db call this function directly? Idk)